crates/sol-types/src/types/enum.rs

```diff
@@ -0,0 +1,36 @@
+use crate::{token::WordToken, Result, SolType, Word};
+use alloc::vec::Vec;
+
+/// Solidity enum. This is always a wrapper around a [`u8`].
+///
+/// ### Implementer's Guide
+///
+/// We do not recommend implementing this trait directly. Instead, we recommend
+/// using the [`sol`][crate::sol] proc macro to parse a Solidity error
+/// definition.
+pub trait SolEnum: Sized + Copy + Into<u8> + TryFrom<u8, Error = crate::Error> {
```

Comment:
enums should `impl SolType` yes?

Reply:
Should they? They're always u8

Reply:
sure, but we can include an enum in a solidity struct, or in function args. so it should be a SolType

Reply:
works fine up until `fn detokenize` because u8 -> Self is fallible:

```rust
impl Encodable<#name> for #name {...}
impl SolType for #name {
    type RustType = #name;

    ...

    fn detokenize(token: Self::TokenType<'_>) -> Self::RustType {
        let x = #uint8_st::detokenize(token);
        todo!("???")
    }
}
```

Reply:
the behavior for `detokenize` an invalid value into a u8 is to ignore all high bytes. should the behavior for an enum be similar? 🤔

Reply:
Maybe with an `__Invalid = u8::MAX` variant for default?

Reply:
That sounds right to me. and then type_check fails for invalid